tenacity/retry.py

```diff
@@ -111,6 +112,61 @@ def __call__(self, attempt):
             return not self.predicate(attempt.result())
 
 
+class retry_if_exception_message(retry_base):
+    """Retries if an exception message equals or matches."""
+
+    def __init__(self, message=None, match=None, require_exception=False):
+        if message and match:
+            raise TypeError(
+                "{}() takes either 'message' or 'match', not both".format(
+                    self.__class__.__name__))
+        self.predicate = self._predicate_factory(message, match)
```

Comment:
just pass the result to the `__init__` method of `retry_if_exception` here (since you will inherit from it). The factory method is not even really needed IMHO, you can put that code in `__init__`.